packages/utils/node/fetch-global-access/lib/fetch-global-access-for-roles.ts

```diff
@@ -0,0 +1,15 @@
+import type { Accountability, GlobalAccess } from '@directus/types';
+import type { Knex } from 'knex';
+import { fetchGlobalAccessForQuery } from '../utils/fetch-global-access-for-query.js';
+
+interface FetchGlobalAccessForRolesContext {
+	knex: Knex;
+}
+
+export async function fetchGlobalAccessForRoles(
+	accountability: Pick<Accountability, 'roles' | 'ip'>,
+	context: FetchGlobalAccessForRolesContext,
+): Promise<GlobalAccess> {
```

Comment:
Since this function and the one it calls doesnt use the complete accountability object do we want to expose the function to end users with its current signature? 🤔

```suggestion
export async function fetchGlobalAccessForRoles(
	roles: string[],
	context: FetchGlobalAccessForRolesContext & { ip?: string }, // `ip` optional as its not required in all cases
): Promise<GlobalAccess> {
```

it would make sense right for the `fetchGlobalAccessForRoles` to take `roles` as its parameter? 🤔

Reply:
Good point, we can build it internally 👍

Reply:
Added in https://github.com/directus/directus/pull/26248/commits/8519430cfef54e8fbe19c6deb925e4b2a56d7847